llvm/include/llvm/Support/Compiler.h

```diff
@@ -114,7 +114,8 @@
 /// this attribute will be made public and visible outside of any shared library
 /// they are linked in to.
 
-#if LLVM_HAS_CPP_ATTRIBUTE(gnu::visibility)
+#if LLVM_HAS_CPP_ATTRIBUTE(gnu::visibility) && defined(__GNUC__) &&            \
+    !defined(__clang__)
```

Comment:
Clang should support this under the GNU standard. Can you add a note on why this condition is needed?

Reply:
I'm not sure what the need for it was, it was part of @tstellar initial changes to the file that i based my code on. I can just remove it if there wasn't some clear purpose for it.

Reply:
Clang supports it, but IIRC the rules about where the function declaration the c++11 style attributes can be placed is different between gcc in clang.  See the comments here:

https://github.com/tstellar/template-visibility/blob/testing/templates.h

Reply:
That file is my pain. I did add a option to the clang tool to switch between the two places where to generate the macro for for functions. I also did try generating export macros on template classes, but it just creates ton a more code that needs to be fixed for MSVC and can be problematic for class static fields when classes with instantiations outside the llvm shared library.